spice_ev/strategies/distributed.py

```diff
@@ -20,14 +19,67 @@ def __init__(self, components, start_time, **kwargs):
-            components, start_time, **strat_options_opps)
+            comps, start_time, **strat_options_opps)
         self.strat_deps = strategy.class_from_str(strat_deps)(
-            components, start_time, **strat_options_deps)
+            comps, start_time, **strat_options_deps)
 
+        # adjust foresight for vehicle events (known one hour in advance)
+        self.A_HORIZON = datetime.timedelta(hours=1)
         # minimum charging time at depot; time to look into the future for prioritization
-        self.C_HORIZON = 3  # min
-        # dict that holds the current vehicles connected to a grid connector for each gc
-        self.v_connect = {gc_id: [] for gc_id in self.world_state.grid_connectors.keys()}
+        self.C_HORIZON = datetime.timedelta(minutes=3)
+        for event in self.events.vehicle_events:
+            event.signal_time = min(event.signal_time, event.start_time-self.A_HORIZON)
+
+        # keep track of connected vehicles per GC (more vehicles might have arrived than CS)
+        self.connected = {gc_id: dict() for gc_id in self.world_state.grid_connectors.keys()}
+        # assumption: one GC each for every station/depot
+        self.virtual_vt = dict()  # virtual vehicle types for stationary batteries
+        self.virtual_cs = dict()  # virtual charging stations for stationary batteries
+        self.strategies = dict()  # tuple of (station type, strategy) for each GC
+        self.gc_battery = dict()  # GC ID -> batteries
+        # set strategy for each GC
+        for cs_id, cs in self.world_state.charging_stations.items():
+            station_type = cs_id.split("_")[-1]
+            prev_type = self.strategies.get(cs.parent)
+            if prev_type is not None:
+                assert prev_type[0] == station_type, f"Station types do not match at {cs.parent}"
+                continue
+            if station_type == "deps":
+                self.strategies[cs.parent] = (station_type, self.strat_deps)
+            elif station_type == "opps":
+                self.strategies[cs.parent] = (station_type, self.strat_opps)
+            else:
+                raise Exception(f"The station {cs.parent} has no declaration such as "
+                                "'opps' or 'deps' attached. Please make sure the "
+                                "ending of the station name is one of the mentioned.")
+
+        # prepare batteries
+        for b_id, bat in self.world_state.batteries.items():
+            # make note to run GC even if no vehicles are connected
+            if self.gc_battery.get(bat.parent):
+                self.gc_battery[bat.parent][b_id] = bat
+            else:
+                self.gc_battery[bat.parent] = {b_id: bat}
+
+            station_type = self.strategies.get(bat.parent)
+            if station_type is None or station_type[0] == "deps":
+                # only batteries at opportunity stations need preparation
+                continue
+            name = f"stationary_{b_id}"
```

Comment:
```suggestion
            name = f"Stationary_BAT_{b_id}"
```
Seems like other variables (at least in the plots) start with a capital letter? Add `BAT` to make it clear that this is a stationary battery.
Same in line 230 and 259